src/components/MovieAdder.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState, ChangeEvent } from "react";
+import AsyncSelect from "react-select/lib/Async";
+import { debounce } from "underscore";
+
+import { MoviePreference, User } from "../types";
+
+const promiseOptions = (inputValue: string) => {
+  const searchParams = new URLSearchParams({ search: inputValue });
+  return fetch("/search-movies?" + searchParams.toString())
+    .then(response => response.json())
+    .then(function(json: {
+      searchResults: Array<{ id: string; title: string }>;
+    }) {
+      return json.searchResults.map(option => ({
+        label: option.title,
+        value: option.id
+      }));
+    });
+};
+
+interface externalMovie {
+  value: number;
+  label: string;
+}
+
+const MovieAdder = (props: { user: User }): JSX.Element => {
+  const [externalMovie, setExternalMovie] = useState<externalMovie | null>(
+    null
+  );
+  const handleChange = (value: any, data: { action: string }) => {
+    if (data.action === "select-option") {
+      setExternalMovie(value as externalMovie);
+    }
+  };
+
+  const clearValue = () => {
+    setExternalMovie(null);
+  };
+
+  const handleAddMoviePreference = () => {
+    if (!externalMovie) {
+      return;
+    }
+
+    fetch(`/users/${props.user.id}/movie-preferences`, {
+      method: "POST",
+      body: JSON.stringify({ externalMovieId: externalMovie.value }),
+      headers: {
+        "Content-Type": "application/json; charset=utf-8"
+      }
+    })
+      .then(response => {
```

Comment:
Doesn't have to be in this PR, but we've written a bunch of fetch logic that could probably start to get consolidated soon

Reply:
Ok I might work on this after errors! Look into axios or something.